agent/consul/state/catalog_test.go

```diff
@@ -962,10 +969,46 @@ func TestStateStore_EnsureService(t *testing.T) {
 		t.Fatalf("bad: %#v", svc)
 	}
 
-	// Index tables were updated.
-	if idx := s.maxIndex("services"); idx != 40 {
+	// Index tables not updated since service did already exist
+	if idx := s.maxIndex("services"); idx != 30 {
 		t.Fatalf("bad index: %d", idx)
 	}
+
+	ns4 := &structs.NodeService{
+		ID:      "service4",
+		Service: "web",
+		Tags:    []string{"prod"},
+		Address: "1.1.1.1",
+		Port:    8000,
+	}
+	if err = s.EnsureService(50, "node1", ns4); err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	if !watchFired(ws) {
+		t.Fatalf("bad")
+	}
+	// Retrieve the service again and ensure it matches..
+	idx, out, err = s.NodeServices(nil, "node1")
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	if idx != 50 {
+		t.Fatalf("bad index: %d", idx)
+	}
+	if out == nil || len(out.Services) != 3 {
+		t.Fatalf("bad: %#v", out)
+	}
+	expect4 := *ns4
+	expect4.CreateIndex, expect4.ModifyIndex = 50, 50
+	if svc := out.Services["service4"]; !reflect.DeepEqual(&expect4, svc) {
+		t.Fatalf("bad: %#v\nVS               : %#v", svc, expect4)
+	}
+
+	// Index tables not updated since service did already exist
+	if idx := s.maxIndex("services"); idx != 50 {
+		t.Fatalf("bad index: %d", idx)
+	}
+
```

Comment:
Might be missing it due to just skimming on GitHub but there doesn't seem to be a test case to cover only tags changing and ensuring that does get a new index?

Reply:
Oops, you are right... adding it now